spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapConfigProperties.java

```diff
@@ -235,23 +151,6 @@ private Stream<NormalizedSource> normalize(String defaultName, String defaultNam
 
 		}
 
-		@Override
```

Comment:
now this in theory is dangerous, because equals/hashCode is based on different parameters in records, while we require only on two of them. It turns out that we never required equals/hashCode, because we get them all as a `List` and transform them into a `List<NormalizedSource>`. And later use wrap `List<NormalizedSource>` into a `Set<NormalizedSource>`, but now all implementations of `NormalizedSource` do have proper hashCode and equals on the fields that we care. So removing these two methods makes sense.

We also have unit tests that validate this behavior.